cli/src/main/java/org/owasp/dependencycheck/App.java

```diff
@@ -296,17 +300,46 @@ private int runScan(String reportDirectory, String[] outputFormats, String appli
     private int determineReturnCode(Engine engine, float cvssFailScore) {
         int retCode = 0;
         //Set the exit code based on whether we found a high enough vulnerability
+        final StringBuilder ids = new StringBuilder();
         for (Dependency d : engine.getDependencies()) {
+            boolean addName = true;
             for (Vulnerability v : d.getVulnerabilities()) {
-                if ((v.getCvssV2() != null && v.getCvssV2().getScore() >= cvssFailScore)
-                        || (v.getCvssV3() != null && v.getCvssV3().getBaseScore() >= cvssFailScore)
-                        || (v.getUnscoredSeverity() != null && SeverityUtil.estimateCvssV2(v.getUnscoredSeverity()) >= cvssFailScore)
-                        || (cvssFailScore <= 0.0f)) { //safety net to fail on any if for some reason the above misses on 0
-                    retCode = 1;
-                    break;
+                final float cvssV2 = v.getCvssV2() != null ? v.getCvssV2().getScore() : -1;
+                final float cvssV3 = v.getCvssV3() != null ? v.getCvssV3().getBaseScore() : -1;
+                final float unscoredCvss = v.getUnscoredSeverity() != null ? SeverityUtil.estimateCvssV2(v.getUnscoredSeverity()) : -1;
+
+                if (cvssV2 >= cvssFailScore
+                        || cvssV3 >= cvssFailScore
+                        || unscoredCvss >= cvssFailScore
+                        //safety net to fail on any if for some reason the above misses on 0
+                        || (cvssFailScore <= 0.0f)) {
+                    String name = v.getName();
+                    if (cvssV3 >= 0.0f) {
+                        name += "(" + cvssV3 + ")";
+                    } else if (cvssV2 >= 0.0f) {
+                        name += "(" + cvssV2 + ")";
+                    } else if (unscoredCvss >= 0.0f) {
+                        name += "(" + unscoredCvss + ")";
+                    }
+                    if (addName) {
+                        addName = false;
+                        ids.append(NEW_LINE).append(d.getFileName()).append(": ");
+                        ids.append(name);
+                    } else {
+                        ids.append(", ").append(name);
+                    }
```

Comment:
Rather than creating all the intermediate strings why not directly append the components to the StringBuilder

```suggestion
                    float score;
                    if (cvssV3 >= 0.0f) {
                        score = cvssV3;
                    } else if (cvssV2 >= 0.0f) {
                        score = cvssV2;
                    } else if (unscoredCvss >= 0.0f) {
                        score = unscoredCvss;
                    }
                    if (addName) {
                        addName = false;
                        ids.append(NEW_LINE).append(d.getFileName()).append(": ");
                        ids.append(v.getName()).append('(').append(score).append(')');
                    } else {
                        ids.append(", ").append(v.getName()).append('(').append(score).append(')');
                    }
```